### PR TITLE
adopt k8s.io/client-go

### LINKED ISSUE #26: adopt k8s.io/client-go

### PR DESCRIPTION
**Pull request by JorritSalverda**
Fixes #26

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -101,37 +107,47 @@ func main() {
...
 						break
 					}
 
-					if *event.Type == k8s.EventAdded || *event.Type == k8s.EventModified || *event.Type == k8s.EventDeleted {
-						if *event.Type == k8s.EventDeleted {
+					if event.Type == watch.Added || event.Type == watch.Modified || event.Type == watch.Deleted {
+
+						service, ok := event.Object.(*v1.Service)
+						if !ok {
+							log.Warn().Msg("Watcher for pods returns event object of incorrect type")
```
Comment (vitorarins) [inline main.go]: typo here? Shouldn't it be "Watcher for services..." ?

File: main.go [hunk 2]
```diff
@@ -101,37 +107,47 @@ func main() {
 		// loop indefinitely
 		for {
 			log.Info().Msg("Watching services for all namespaces...")
-			watcher, err := client.CoreV1().WatchServices(context.Background(), k8s.AllNamespaces)
+			timeoutSeconds := int64(300)
+			watcher, err := kubeClientset.CoreV1().Services("").Watch(metav1.ListOptions{
+				TimeoutSeconds: &timeoutSeconds,
+			})
```
Comment (vitorarins) [inline main.go]: in letsencrypt controller there is: `defer watcher.Close()` right after creation.
Should we do something similar for this? Maybe close the watcher's result channel

File: main.go (hunk 1 above)
Comment (JorritSalverda) [reply main.go]: Heh heh, serves me for copying this from elsewhere...

File: main.go (hunk 2 above)
Comment (JorritSalverda) [reply main.go]: This watcher no longer has a `Close` method.

**Review by vitorarins — APPROVED** [review]

**Pull request merged**